mlir/lib/Conversion/FuncToEmitC/FuncToEmitC.cpp

```diff
@@ -0,0 +1,119 @@
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file implements patterns to convert the Func dialect to the EmitC
+// dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#include "mlir/Conversion/FuncToEmitC/FuncToEmitC.h"
+
+#include "mlir/Dialect/EmitC/IR/EmitC.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Transforms/DialectConversion.h"
+
+using namespace mlir;
+
+//===----------------------------------------------------------------------===//
+// Conversion Patterns
+//===----------------------------------------------------------------------===//
+
+namespace {
+class CallOpConversion final : public OpConversionPattern<func::CallOp> {
+public:
+  using OpConversionPattern<func::CallOp>::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(func::CallOp callOp, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    // multiple results func was not converted to spirv.func
+    if (callOp.getNumResults() > 1)
+      return rewriter.notifyMatchFailure(
+          callOp, "Only functions with zero or one result can be converted");
+
+    rewriter.replaceOpWithNewOp<emitc::CallOp>(
+        callOp,
+        callOp.getNumResults() ? callOp.getResult(0).getType() : nullptr,
+        adaptor.getOperands(), callOp->getAttrs());
+
+    return success();
+  }
+};
+
+class FuncOpConversion final : public OpConversionPattern<func::FuncOp> {
+public:
+  using OpConversionPattern<func::FuncOp>::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(func::FuncOp funcOp, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+
+    if (funcOp.getFunctionType().getNumResults() > 1)
+      return rewriter.notifyMatchFailure(
+          funcOp, "Only functions with zero or one result can be converted");
+
+    if (funcOp.isDeclaration())
+      return rewriter.notifyMatchFailure(funcOp,
+                                         "Declarations cannot be converted");
+
+    // Create the converted emitc.func op.
+    emitc::FuncOp newFuncOp = rewriter.create<emitc::FuncOp>(
+        funcOp.getLoc(), funcOp.getName(), funcOp.getFunctionType());
+
+    // Copy over all attributes other than the function name and type.
+    for (const auto &namedAttr : funcOp->getAttrs()) {
+      if (namedAttr.getName() != funcOp.getFunctionTypeAttrName() &&
+          namedAttr.getName() != SymbolTable::getSymbolAttrName())
+        newFuncOp->setAttr(namedAttr.getName(), namedAttr.getValue());
+    }
+
+    // Create add `static` to specifiers if `func.func` is private.
+    if (funcOp.isPrivate()) {
+      StringAttr specifier = rewriter.getStringAttr("static");
+      ArrayAttr specifiers = rewriter.getArrayAttr(specifier);
+      newFuncOp.setSpecifiersAttr(specifiers);
```

Comment:
There is `rewriter.getStrArrayAttr`. So something like `rewriter.getStrArrayAttr({"static"})` should work.